app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModel.kt

```diff
@@ -103,6 +111,42 @@ class HomeViewModel @Inject constructor(
         _showTrendingCoinsLoading.value = false
     }
 
+    @OptIn(FlowPreview::class)
+    fun refresh() = execute {
+        _isRefreshing.value = true
+
+        delay(5000L)
+
+        val getMyCoins = getMyCoinsUseCase.execute(
+            GetMyCoinsUseCase.Input(
+                currency = FIAT_CURRENCY,
+                order = MY_COINS_ORDER,
+                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
+                itemPerPage = MY_COINS_ITEM_PER_PAGE,
+                page = MY_COINS_INITIAL_PAGE
+            )
+        )
+
+        val getTrendingCoins = getTrendingCoinsUseCase.execute(
+            GetTrendingCoinsUseCase.Input(
+                currency = FIAT_CURRENCY,
+                order = MY_COINS_ORDER,
+                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
+                itemPerPage = MY_COINS_ITEM_PER_PAGE,
+                page = MY_COINS_INITIAL_PAGE
+            )
+        )
+
+        getMyCoins.zip(getTrendingCoins) { myCoins, trendingCoins ->
+            _myCoins.emit(myCoins.map { it.toUiModel() })
+            _trendingCoins.emit(trendingCoins.map { it.toUiModel() })
+        }.catch { e ->
+            _error.emit(e)
+        }.collect()
+
+        _isRefreshing.value = false
```

Comment:
Then I don't think we even need zip here, because the original logic of `showLoading` will countdown automatically in asynchronous way. The `false` value will be emitted only when `loadingCount = 0` 🤔

Ref here:  https://github.com/nimblehq/jetpack-compose-crypto/blob/e7de43fcafe97d8b9bb4e407335ea33a34ea19f6/app/src/main/java/co/nimblehq/compose/crypto/ui/base/BaseViewModel.kt#L50-L55

Reply:
@luongvo I intend to apply `zip` operator to emit 2 lists at the same time rather than emit `myCoins` then wait for `trendingCoins` complete to refresh the list